Framework/CurveFitting/test/Functions/ProcessBackgroundTest.h

```diff
@@ -342,23 +319,155 @@ class ProcessBackgroundTest : public CxxTest::TestSuite {
+    }
+
+    MersenneTwister mt(1234, 0.0, 1000000.0);
+    std::vector<double> bkgdpts(10000);
+    for (auto it = bkgdpts.begin(); it != bkgdpts.end(); it++) {
+      *it = mt.nextValue();
+    }
+
+    pb3.initialize();
+    pb3.setProperty("InputWorkspace", inpws);
+    pb3.setProperty("OutputWorkspace", "SelectedBackgroundPoints");
+    pb3.setProperty("Options", "SelectBackgroundPoints");
+    pb3.setProperty("BackgroundPointSelectMode",
+                    "Input Background Points Only");
+    pb3.setProperty("SelectionMode", "FitGivenDataPoints");
+    pb3.setProperty("BackgroundType", "Polynomial");
+    pb3.setProperty("BackgroundPoints", bkgdpts);
+    pb3.setProperty("WorkspaceIndex", 0);
+    pb3.setProperty("NoiseTolerance", 100.0);
+
+    // 4. Set up select background from input function test
+    auto dataws = createInputWS("DiffractionData2", 50000, 50000);
+    for (size_t i = 0; i < 50000; ++i) {
+      dataws->mutableX(0)[i] = double(i);
+      dataws->mutableY(0)[i] =
+          double(i) * double(i) + sin(double(i) / 180. * 3.14);
+    }
+
+    // Create background function
+    TableWorkspace_sptr functablews = boost::make_shared<TableWorkspace>();
+    functablews->addColumn("str", "Name");
+    functablews->addColumn("double", "Value");
+    TableRow row0 = functablews->appendRow();
+    row0 << "A0" << 0.;
+    TableRow row1 = functablews->appendRow();
+    row1 << "A1" << 0.;
+    TableRow row2 = functablews->appendRow();
+    row2 << "A2" << 1.;
+    AnalysisDataService::Instance().addOrReplace("BackgroundParameters",
+                                                 functablews);
+
+    // Create and set up algorithm
+    pb4.initialize();
+    pb4.setProperty("InputWorkspace", dataws);
+    pb4.setProperty("WorkspaceIndex", 0);
+    pb4.setProperty("OutputWorkspace", "SelectedBackgroundPoints2");
+    pb4.setProperty("Options", "SelectBackgroundPoints");
+    pb4.setProperty("BackgroundType", "Polynomial");
+    pb4.setProperty("SelectionMode", "UserFunction");
+    pb4.setProperty("BackgroundTableWorkspace", functablews);
+    pb4.setProperty("OutputBackgroundParameterWorkspace",
+                    "OutBackgroundParameters");
+    pb4.setProperty("UserBackgroundWorkspace", "VisualWS");
+    pb4.setProperty("OutputBackgroundType", "Chebyshev");
+    pb4.setProperty("OutputBackgroundOrder", 6);
+    pb4.setProperty("NoiseTolerance", 0.25);
+  }
+
+  void tearDown() override {
+    AnalysisDataService::Instance().remove("Background1");
+    AnalysisDataService::Instance().remove("Background2");
+    AnalysisDataService::Instance().remove("RefBackground");
+    AnalysisDataService::Instance().remove("DiffractionData1");
+    AnalysisDataService::Instance().remove("DiffractionData2");
+    AnalysisDataService::Instance().remove("BackgroundParameters");
+  }
+
+  void testPerformanceWS() {
+    pb1.execute();
+    pb2.execute();
+    pb3.execute();
+    pb4.execute();
```

Comment:
Have you considered putting these into 4 separate tests so we can see the timings of the individual runs?